test/one/profiler/test/SourceLocator.java

```diff
@@ -0,0 +1,68 @@
+package one.profiler.test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class SourceLocator {
+    public static String tryGetFrom(Throwable e, int ignoreFrames) {
+        StackTraceElement[] stackTrace = e.getStackTrace();
+
+        if (stackTrace.length > ignoreFrames) {
+            StackTraceElement element = stackTrace[ignoreFrames];
+            String fileName = element.getFileName();
+            int lineNumber = element.getLineNumber();
+            String className = element.getClassName();
+
+            try {
+                Class<?> clazz = Class.forName(className);
```

Comment:
Why do we need to instantiate a class just to convert its name to a path? Isn't `className` sufficient?

Reply:
We already had `fileName` with extension, simplifying this.